### PR TITLE
[EDL][Players] Cleanup FIXMEs

### PR DESCRIPTION
**Pull request by enen92**
## Description
This provides a few assorted improvements to the EDL interface (mostly leftovers):

- Remove passing of const values
- Remove default parameters
- Remove using random bool variables instead of enum classes
- Remove some hungarian notation

I don't want it to grow bigger than it's already so please consider other stuff out of the scope.

### CODE REVIEW

**Review by howie-f — APPROVED** [review]
nice work, i would only consider making the enum classes type `bool`

**Review by fuzzard — APPROVED** [review]

File: xbmc/cores/Direction.h [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+#pragma once
+
+/*!
+ * @brief Specifies/Abstracts a direction
+*/
+enum class Direction : bool
+{
+  FORWARD, /*!< Forward */
+  BACKWARD /*!< Backwards */
```
Comment (ksooo) [inline xbmc/cores/Direction.h:17]: Remove the s from comment?

File: xbmc/cores/VideoPlayer/Edl.cpp [hunk 2]
```diff
@@ -882,37 +884,37 @@ EDL::Action CEdl::GetLastEditActionType() const
   return m_lastEditActionType;
 }
 
-bool CEdl::GetNextSceneMarker(bool bPlus, const int iClock, int *iSceneMarker)
+bool CEdl::GetNextSceneMarker(Direction direction, int clock, int* sceneMarker)
```
Comment (ksooo) [inline xbmc/cores/VideoPlayer/Edl.cpp:887]: Would it be possible to replace the int pointer with a reference?

**Review by ksooo — COMMENTED** [review]
Nice!

File: xbmc/cores/VideoPlayer/Edl.cpp (hunk 2 above)
Comment (enen92) [reply xbmc/cores/VideoPlayer/Edl.cpp]: I was not really interested in doing more in this PR but done in https://github.com/xbmc/xbmc/pull/25346/commits/0a4fb522bf2468697d9281c6c2480257cd309f19

**Review by ksooo — APPROVED** [review]
lgtm

**Pull request merged**